### PR TITLE
220166_Align-Invoice-API-Models-InvoiceLine

### PR DESCRIPTION
**Pull request by fshehzad-defra**
This is part one of estimated 3 PR's

- Rename Invoice to PaymentRequestsBatch
- Rename InvoiceHeader to PaymentRequest
- Align InvoiceLine model to spec

### CODE REVIEW

File: EST.MIT.Invoice.Api.Test/InvoiceLineRequiredFieldsTests.cs [hunk 1]
```diff
@@ -75,6 +78,29 @@ public void Test_SchemeCode_For_Null()
...
 
+        [Theory]
+        [InlineData(2020)]
+        [InlineData(3000)]
+        public void Test_Marketing_Year_For_Value_Not_In_Range(int marketingYear)
+        {
+            //Arrange
+            InvoiceLine invoiceLine = new InvoiceLine()
+            {
+                Currency = "£",
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/InvoiceLineRequiredFieldsTests.cs:89]: Should currency be "GBP" ? (It's GBP or EUR)

File: EST.MIT.Invoice.Api.Test/InvoiceLineValidatorTests.cs [hunk 2]
```diff
@@ -369,8 +380,9 @@ public async Task Given_InvoiceLine_When_Value_Is_Equal_To_Zero_Then_InvoiceLine
             response.ShouldNotHaveValidationErrorFor(x => x.Description);
             response.ShouldNotHaveValidationErrorFor(x => x.FundCode);
             response.ShouldNotHaveValidationErrorFor(x => x.Currency);
+            response.ShouldNotHaveValidationErrorFor(x => x.MarketingYear);
             Assert.Single(response.Errors);
-            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("Invoice line value must be non-zero")) == 1);
+            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("PaymentRequestsBatch line value must be non-zero")) == 1);
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/InvoiceLineValidatorTests.cs:385]: Probably a search/replace error. Error messages should stick with "Invoice". Looks like other messages affected.

File: EST.MIT.Invoice.Api.Test/InvoiceLineValidatorTests.cs [hunk 3]
```diff
@@ -342,8 +351,9 @@ public async Task Given_InvoiceLine_When_Value_Has_More_Than_2DP_Then_InvoiceLin
             response.ShouldNotHaveValidationErrorFor(x => x.Description);
             response.ShouldNotHaveValidationErrorFor(x => x.FundCode);
             response.ShouldNotHaveValidationErrorFor(x => x.Currency);
+            response.ShouldNotHaveValidationErrorFor(x => x.MarketingYear);
             Assert.Single(response.Errors);
-            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("Invoice line value cannot be more than 2dp")) == 1);
+            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("PaymentRequestsBatch line value cannot be more than 2dp")) == 1);
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/InvoiceLineValidatorTests.cs:356]: Invoice text shouldn't have been changed here.

File: EST.MIT.Invoice.Api.Test/PaymentRequestValidatorCustomerIdTest.cs [hunk 4]
```diff
@@ -197,7 +199,7 @@ public async Task Given_InvoiceHeader_When_SBI_FRN_And_VendorId_Supplied_Then_In
             response.ShouldNotHaveValidationErrorFor(x => x.AppendixReferences);
 
             Assert.Single(response.Errors);
-            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("Invoice must only have Single Business Identifier (SBI), Firm Reference Number (FRN) or Vendor ID")) == 1);
+            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("PaymentRequestsBatch must only have Single Business Identifier (SBI), Firm Reference Number (FRN) or Vendor ID")) == 1);
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/PaymentRequestValidatorCustomerIdTest.cs:202]: Should remain Invoice for the error message. Search and replaced?

File: EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs [hunk 5]
```diff
@@ -488,15 +490,15 @@ public async Task Given_InvoiceHeader_When_Value_Has_More_Than_2DP_Then_InvoiceH
             response.ShouldNotHaveValidationErrorFor(x => x.AppendixReferences);
 
             Assert.Equal(2, response.Errors.Count);
-            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("Invoice value cannot be more than 2dp")) == 1);
-            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("Invoice line value cannot be more than 2dp")) == 1);
+            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("PaymentRequestsBatch value cannot be more than 2dp")) == 1);
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs:493]: Should remain Invoice. Search and replace error?

File: EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs [hunk 6]
```diff
@@ -540,8 +543,8 @@ public async Task Given_InvoiceHeader_When_Value_Is_Equal_To_Zero_Then_InvoiceHe
             response.ShouldNotHaveValidationErrorFor(x => x.InvoiceLines);
             response.ShouldNotHaveValidationErrorFor(x => x.AppendixReferences);
             Assert.Equal(2, response.Errors.Count);
-            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("Invoice value must be non-zero")) == 1);
-            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("Invoice line value must be non-zero")) == 1);
+            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("PaymentRequestsBatch value must be non-zero")) == 1);
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs:546]: Should remain Invoice.

File: EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs [hunk 7]
```diff
@@ -607,7 +612,7 @@ public async Task Given_InvoiceHeader_When_Value_Does_Not_Equal_Sum_Of_InvoiceLi
             response.ShouldNotHaveValidationErrorFor(x => x.AppendixReferences);
 
             Assert.Single(response.Errors);
-            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains($"Invoice Value ({invoiceValue}) does not equal the sum of Line Values ({invoiceLine1Value + invoiceLine2Value}")) == 1);
+            Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains($"PaymentRequestsBatch Value ({invoiceValue}) does not equal the sum of Line Values ({invoiceLine1Value + invoiceLine2Value}")) == 1);
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs:615]: Should remain Invoice

File: EST.MIT.Invoice.Api/Endpoints/InvoiceDeleteEndpoints.cs [hunk 8]
```diff
@@ -21,7 +21,7 @@ public static IEndpointRouteBuilder MapInvoiceDeleteEndpoints(this IEndpointRout
     public static async Task<IResult> DeleteInvoice(string id, string scheme, ICosmosService cosmosService, IEventQueueService eventQueueService)
     {
         await cosmosService.Delete(id, scheme);
-        await eventQueueService.CreateMessage(id, "deleted", "invoice-deleted", "Invoice updated");
+        await eventQueueService.CreateMessage(id, "deleted", "invoice-deleted", "PaymentRequestsBatch updated");
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Endpoints/InvoiceDeleteEndpoints.cs:24]: Should remain Invoice. Search and replace error.

File: EST.MIT.Invoice.Api.Test/InvoiceLineRequiredFieldsTests.cs (hunk 1 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api.Test/InvoiceLineRequiredFieldsTests.cs:89]: Its just mocked data, doesnt really matter what it is as long as it makes sense.

File: EST.MIT.Invoice.Api.Test/InvoiceLineValidatorTests.cs (hunk 2 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api.Test/InvoiceLineValidatorTests.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api.Test/InvoiceLineValidatorTests.cs (hunk 3 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api.Test/InvoiceLineValidatorTests.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api/Endpoints/InvoicePostEndpoints.cs [hunk 9]
```diff
@@ -54,15 +54,15 @@ public static async Task<IResult> CreateBulkInvoices(BulkInvoices invoices, IVal
 
         if (!validationResult.IsValid)
         {
-            await eventQueueService.CreateMessage(reference, "invalid", "bulk-invoice-validation-falied", "Bulk invoice validation failed");
+            await eventQueueService.CreateMessage(reference, "invalid", "bulk-paymentRequestsBatch-validation-falied", "Bulk paymentRequestsBatch validation failed");
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Endpoints/InvoicePostEndpoints.cs]: Hmmm. should probably remain "invoice" ...?

File: EST.MIT.Invoice.Api.Test/PaymentRequestValidatorCustomerIdTest.cs (hunk 4 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api.Test/PaymentRequestValidatorCustomerIdTest.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs (hunk 5 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs (hunk 6 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api/Endpoints/InvoicePostEndpoints.cs [hunk 10]
```diff
@@ -54,15 +54,15 @@ public static async Task<IResult> CreateBulkInvoices(BulkInvoices invoices, IVal
...
             return Results.ValidationProblem(validationResult.ToDictionary());
         }
 
         var bulkInvoiceCreated = await cosmosService.CreateBulk(invoices);
 
         if (bulkInvoiceCreated is null)
         {
-            await eventQueueService.CreateMessage(reference, "failed", "bulk-invoice-creation-falied", "Bulk invoice creation failed");
+            await eventQueueService.CreateMessage(reference, "failed", "bulk-paymentRequestsBatch-creation-falied", "Bulk paymentRequestsBatch creation failed");
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Endpoints/InvoicePostEndpoints.cs]: should probably remain 'invoice'

File: EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs (hunk 7 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api.Test/PaymentRequestValidatorTest.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api/Endpoints/InvoiceDeleteEndpoints.cs (hunk 8 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Endpoints/InvoiceDeleteEndpoints.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api/Endpoints/InvoicePostEndpoints.cs (hunk 9 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Endpoints/InvoicePostEndpoints.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api/Endpoints/InvoicePostEndpoints.cs (hunk 10 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Endpoints/InvoicePostEndpoints.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs [hunk 11]
```diff
@@ -11,40 +11,40 @@ public static class InvoicePutEndpoints
     [ExcludeFromCodeCoverage]
     public static IEndpointRouteBuilder MapInvoicePutEndpoints(this IEndpointRouteBuilder app)
     {
-        app.MapPut("/invoice/{invoiceId}", UpdateInvoice)
+        app.MapPut("/paymentRequestsBatch/{invoiceId}", UpdateInvoice)
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs:14]: should probably remain 'invoice'

File: EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs [hunk 12]
```diff
@@ -11,40 +11,40 @@ public static class InvoicePutEndpoints
...
 
         if (invoiceUpdated is null)
         {
-            await eventQueueService.CreateMessage(invoice.Id, invoice.Status, "invoice-update-falied", "Invoice update failed", invoice);
+            await eventQueueService.CreateMessage(paymentRequestsBatch.Id, paymentRequestsBatch.Status, "paymentRequestsBatch-update-falied", "PaymentRequestsBatch update failed", paymentRequestsBatch);
             return Results.BadRequest();
         }
 
-        await eventQueueService.CreateMessage(invoice.Id, invoice.Status, "invoice-update", "Invoice updated", invoice);
+        await eventQueueService.CreateMessage(paymentRequestsBatch.Id, paymentRequestsBatch.Status, "paymentRequestsBatch-update", "PaymentRequestsBatch updated", paymentRequestsBatch);
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs]: The text bits probably should stay 'invoice'

File: EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs [hunk 13]
```diff
@@ -11,40 +11,40 @@ public static class InvoicePutEndpoints
...
 
-        if (invoice.Status == InvoiceStatuses.Approved)
+        if (paymentRequestsBatch.Status == InvoiceStatuses.Approved)
         {
-            var message = JsonSerializer.Serialize(new InvoiceGenerator { Id = invoice.Id, Scheme = invoice.SchemeType });
+            var message = JsonSerializer.Serialize(new InvoiceGenerator { Id = paymentRequestsBatch.Id, Scheme = paymentRequestsBatch.SchemeType });
             await queueService.CreateMessage(message);
-            await eventQueueService.CreateMessage(invoice.Id, invoice.Status, "invoice-payment-request-sent", "Invoice payment request sent");
+            await eventQueueService.CreateMessage(paymentRequestsBatch.Id, paymentRequestsBatch.Status, "paymentRequestsBatch-payment-request-sent", "PaymentRequestsBatch payment request sent");
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs]: the text bits should probably remain 'invoice'

File: EST.MIT.Invoice.Api/Endpoints/SwaggerDefinition.cs [hunk 14]
```diff
@@ -15,6 +15,6 @@ public static void SwaggerEndpoints(this WebApplication app)
     public static void AddSwaggerServices(this IServiceCollection services)
     {
         services.AddEndpointsApiExplorer();
-        services.AddSwaggerGen(c => c.SwaggerDoc("v1", new OpenApiInfo { Title = "ManualInvoiceTemplatesApi", Version = "v1", Description = "Invoice api for manual invoice templates" }));
+        services.AddSwaggerGen(c => c.SwaggerDoc("v1", new OpenApiInfo { Title = "ManualInvoiceTemplatesApi", Version = "v1", Description = "PaymentRequestsBatch api for manual invoice templates" }));
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Endpoints/SwaggerDefinition.cs]: This should remain invoice in description

File: EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs (hunk 11 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs:14]: Good spot, fixed

File: EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs (hunk 12 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api/Endpoints/SwaggerDefinition.cs (hunk 14 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Endpoints/SwaggerDefinition.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs (hunk 13 above)
Comment (fshehzad-defra) [reply EST.MIT.Invoice.Api/Endpoints/InvoicePutEndpoints.cs]: Good spot, fixed

File: EST.MIT.Invoice.Api.Test/InvoiceLineRequiredFieldsTests.cs [hunk 15]
```diff
@@ -47,7 +48,8 @@ public void Test_Description_For_Null()
                 Currency = "£",
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/InvoiceLineRequiredFieldsTests.cs:48]: I know this wasn't changed, and is only a test, but should be GBP or EUR

File: EST.MIT.Invoice.Api.Test/InvoiceLineRequiredFieldsTests.cs (hunk 1 above)
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/InvoiceLineRequiredFieldsTests.cs:89]: GBP

**Review by andygibbonsdefra — APPROVED** [review]
Looks Good. Only spotted £=>GBP in the tests, but that's not critical.

**Pull request merged**